relay/relay.go

```diff
@@ -510,7 +521,32 @@ func (r *Relay) reSetupMeta(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	err = r.meta.AddDir(uuid, nil, nil)
+
+	var newPos *mysql.Position
+	var newGset gtid.Set
+	var newUUIDSufiix string
+	if len(r.cfg.UUIDSuffix) != 0 {
+		// if bound or rebound to a source, clear all relay log and meta
+		if err = r.PurgeRelayDir(); err != nil {
+			return err
+		}
+		r.ResetMeta()
+
+		newUUIDSufiix = r.cfg.UUIDSuffix
+		// reset the UUIDSuffix
+		r.cfg.UUIDSuffix = ""
```

Comment:
where did `r.cfg.UUIDSuffix` set to a not empty value?

Reply:
copy from https://github.com/pingcap/dm/blob/8f7cf97c49ad3749f4d7bbd4e9fc075d0957b6b1/relay/config.go#L65 which set here https://github.com/pingcap/dm/blob/8f7cf97c49ad3749f4d7bbd4e9fc075d0957b6b1/dm/worker/server.go#L559